### ISSUE TITLE
Carthage build fails on latest release (2.0.2)

### ISSUE THREAD

**Issue by rpassis** [Leanplum/Leanplum-iOS-SDK#53]
When trying to update to the latest release, after running 

`carthage update Leanplum-iOS-SDK`

The build fails with the following error message:

````
*** Skipped installing Leanplum-iOS-SDK.framework binary due to the error:                                                                                     │@@ -7,3 +7,10 @@
        "Failed to read file or folder at /private/var/folders/v9/b69j6j112xjbggkzm4xpkm3h0000gn/T/carthage-archive.ghwvVH/__MACOSX/Leanplum.framework: Error D│ //
omain=com.antitypical.Result Code=0 "the DTSDKName key in its plist file is missing" UserInfo={com.antitypical.Result.file=/tmp/carthage-20170710-56241-9g56uj/│ 
Source/CarthageKit/Project.swift, com.antitypical.Result.function=readFailed, com.antitypical.Result.line=1100, NSLocalizedDescription=the DTSDKName key in its│ import Foundation
 plist file is missing}"                                                                                                                                       │+
*** Checking out Leanplum-iOS-SDK at "2.0.2"                                                                                                                   │+class Leanplum {
*** xcodebuild output can be found in /var/folders/v9/b69j6j112xjbggkzm4xpkm3h0000gn/T/carthage-xcodebuild.x4OSkr.log                                          │+
*** Skipped building Leanplum-iOS-SDK due to the error:                                                                                                        │+  static func bootstrap() {
Dependency "Leanplum-iOS-SDK" has no shared framework schemes                                                                                                  │+  }
````

Version 2.0.1 and below works as expected.

Cheers,

Rog

**Comment by benmarten** [comment]
Thats weird. I haven't tried it myself yet, but it seems to me that Carthage is looking for an xcschme `.xcodeproj/xcshareddata/xcschemes`...
What's odd is that it should just be using the framework file instead....

Similar issues:
- https://github.com/Carthage/Carthage/issues/1056
- https://github.com/Carthage/Carthage/issues/1109

We will need to dig deeper into this...

**Comment by benmarten** [comment]
I am able to reproduce the issue, but I have no clue what's going on. I've opened this ticket on the Carthage repo, to see if there's anything we can do to investigate the issue more deeply and find the root cause: https://github.com/Carthage/Carthage/issues/2084

**Comment by rpassis** [comment]
I'm not sure either but it looks like it might be an issue with the `Result` dependency you guys currently use. 

Perhaps see if the latest Leanplum build updated that dependency and go from there? 

Here's the repo for Result https://github.com/antitypical/Result

**Comment by ikesyo** [comment]
Could you please check this?

https://github.com/Carthage/Carthage/issues/1776#issuecomment-290451271

**Comment by rpassis** [comment]
I checked it myself and it doesn't seem to be the case.

**Comment by ikesyo** [comment]
This is actually the case:

```shell
$ unzip -l Leanplum.framework.zip
Archive:  Leanplum.framework.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  07-21-2017 03:21   Leanplum.framework/
     6148  07-21-2017 03:21   Leanplum.framework/.DS_Store
        0  07-21-2017 03:21   __MACOSX/
        0  07-21-2017 03:21   __MACOSX/Leanplum.framework/
      120  07-21-2017 03:21   __MACOSX/Leanplum.framework/._.DS_Store
        0  07-19-2017 06:02   Leanplum.framework/Headers/
    35085  07-19-2017 06:02   Leanplum.framework/Headers/Leanplum.h
        0  07-21-2017 03:21   __MACOSX/Leanplum.framework/Headers/
      222  07-19-2017 06:02   __MACOSX/Leanplum.framework/Headers/._Leanplum.h
     4958  07-19-2017 06:02   Leanplum.framework/Headers/LPInbox.h
      222  07-19-2017 06:02   __MACOSX/Leanplum.framework/Headers/._LPInbox.h
      222  07-19-2017 06:02   __MACOSX/Leanplum.framework/._Headers
      737  07-19-2017 06:02   Leanplum.framework/Info.plist
      222  07-19-2017 06:02   __MACOSX/Leanplum.framework/._Info.plist
  9923080  07-19-2017 06:02   Leanplum.framework/Leanplum
      222  07-19-2017 06:02   __MACOSX/Leanplum.framework/._Leanplum
        0  07-21-2017 03:21   Leanplum.framework/Modules/
       97  07-21-2017 03:21   Leanplum.framework/Modules/module.modulemap
        0  07-21-2017 03:21   __MACOSX/Leanplum.framework/Modules/
      328  07-21-2017 03:21   __MACOSX/Leanplum.framework/Modules/._module.modulemap
      222  07-21-2017 03:21   __MACOSX/Leanplum.framework/._Modules
      222  07-21-2017 03:21   __MACOSX/._Leanplum.framework
---------                     -------
  9972107                     22 files
```

**Comment by rpassis** [comment]
Thanks @ikesyo not sure why I didn’t notice it when I tried.

**Comment by benmarten** [comment]
Thanks, I fixed it manually... Verified `carthage update` works again...
I modified our build scripts: `zip -r9 "Leanplum-$IOS_VERSION_STRING.zip" ./* -x ./*.zip -x "*.DS_Store" -x "__MACOSX/" -x "__MACOSX/*"`
This should prevent it from happening again...

**Comment by rpassis** [comment]
Yup can confirm this is fixed for this 2.0.2. Thanks all!
